src/Authoring/WinRT.SourceGenerator/CsWinRTDiagnosticStrings.resx

```diff
@@ -369,4 +369,16 @@
   <data name="ComImportInterfaceCast_Text" xml:space="preserve">
     <value>The type '{0}' used in a cast operation is a [ComImport] interface, which is not compatible with CsWinRT objects in AOT scenarios. Consider using the COM generators to define the interface, or manually handling the interface query on the underlying native object.</value>
   </data>
+  <data name="RuntimeClassCast_Brief" xml:space="preserve">
+    <value>Casts to WinRT runtime class types are not trim-safe</value>
+  </data>
+  <data name="RuntimeClassCast_Text" xml:space="preserve">
```

Comment:
I feel like both error messages, but especially this one, is missing the context on when it is not trim-safe / might be trimmed which is when they are declared as object / lacking static type information.

Reply:
Well the problem is that it's more complicated than that. It's not just when there's no static type information. Casting to a derived class is also not trim-safe, even if the source type is not just `object`. I can update the messages and include this info too, was just not sure how verbose and long we wanted the messages to be? 😅

Reply:
Well I would still consider the latter to be lack of static type information but with respect to not being aware of derived type.  We can probably generalize it with something along the lines of can't be determined with static type information or so?

Reply:
How about

Casting to type '{0}' where the type being cast to is never instantiated in the C# implementation is not trim-safe as the type can be trimmed.

or

Casting to type '{0}' where the type being cast to was statically not known about when the C# wrapper was created is not trim-safe as the type can be trimmed.